### PR TITLE
ENH: Load ruler annotations

### LINKED ISSUE #118: Ruler support for '-n' load annotations
Currently only the Box annotation is supported by ImageViewer for loading annotations.  Add support for Ruler annotations (note, the export of Ruler annotations currently works).

See: https://github.com/KitwareMedical/ImageViewer/blob/db6c8b40e370fec17812835de80c665c0ef4a6d9/QtImageViewer/QtImageViewer.cxx#L324-L372

### PR DESCRIPTION
**Pull request by tao558**
Load in ruler annotations. Resolves #118 

Other changes:
* Write out ruler annotation metadata to json
* Use the metadata to reinitialize the state of the onsd and rainbow metadata generators, so the user can pick up where they left off after reading in annotations

TODO:
- [x] Should we always show the ruler annotations when viewing overlay data? Similar to https://github.com/KitwareMedical/ImageViewer/pull/101/commits/4956c6882e0825293181fb861a8e29fa646031fb. What if the user shows ruler and box annotations at the same time?

### CODE REVIEW

**Comment by tao558** [comment]
@brad-t-moore @aylward This is all set. Would one of you please take a look at your earliest convenience? Thank you.

File: QtImageViewer/QtImageViewer.cxx [hunk 1]
```diff
@@ -318,85 +318,205 @@ bool QtImageViewer::loadJSONAnnotations(QString filePathToLoad)
     return false;
   }
 
-  bool status = false;
+  bool status = true;
```
Comment (brad-t-moore) [inline QtImageViewer/QtImageViewer.cxx:321]: Does this change make sense?  Seems like the following equal-or calls are now always true?

**Comment by brad-t-moore** [comment]
LGTM after the review comment about `status` is addressed.  I think we probably need to add an issue to make rulers render even if ruler mode isn't selected.

File: QtImageViewer/QtImageViewer.cxx (hunk 1 above)
Comment (tao558) [reply QtImageViewer/QtImageViewer.cxx]: You're absolutely correct, I switched it from `false` to `true` in that commit, then switched the `|=` to `&=`, then realized that didn't make sense either. I reverted the change but forgot to switch `status` back. See updated file

Comment (brad-t-moore) [reply QtImageViewer/QtImageViewer.cxx]: Looks good!

**Comment by tao558** [comment]
> I think we probably need to add an issue to make rulers render even if ruler mode isn't selected.

This change is a one-liner. I can push it to this branch, if you want?

**Comment by brad-t-moore** [comment]
@tao558 Sure, if you want to add it I can test it within the hour.

-Brad

**Comment by tao558** [comment]
> @tao558 Sure, if you want to add it I can test it within the hour.

All set whenever you're ready.

File: QtImageViewer/QtGlSliceView.cxx [hunk 2]
```diff
@@ -2536,6 +2535,7 @@ void QtGlSliceView::paintGL( void )
     {
     glEnable(GL_BLEND);
     getBoxToolCollection()->paint();
+    getRulerToolCollection()->paint();
```
Comment (brad-t-moore) [inline QtImageViewer/QtGlSliceView.cxx:2538]: I can confirm the rulers now stay rendered when you move off the tool.  They don't load right away though, i.e., you have to toggle through the ruler first to get it to render.  Is that something that can be added at load?

Comment (tao558) [reply QtImageViewer/QtGlSliceView.cxx:2538]: Yes, see updated commit https://github.com/KitwareMedical/ImageViewer/pull/121/commits/e3f2d116fae833645586c8404f19a8de5703e32c

**Comment by brad-t-moore** [comment]
@tao558 Great! LGTM!

**Comment by tao558** [comment]
> @tao558 Great! LGTM!

Great. I cant merge as I don't have the permissions, but I'm all set with this PR. Likewise with #122

**Pull request merged**